pep-0536.txt

```diff
@@ -0,0 +1,184 @@
+
+    >>> f'''A complex trick: {
+    ... bag['bag']  # recursive bags!
+    ... }'''
+    SyntaxError: f-string expression part cannot include '#'
+
+#. Expression portions need to wrap ``':'`` and ``'!'`` in braces::
+
+    >>> f'Useless use of lambdas: { lambda x: x*2 }'
+    SyntaxError: unexpected EOF while parsing
+
+These limitations are non-obvious and counter-intuitive unless the user is
+familiar with the f-literals’ implementation details.  Furthermore,
+they serve no purpose from a language user perspective and can be lifted
+by giving f-literals a regular grammar without exceptions
+and implementing it using dedicated parse code.
+
+.. note:: put preceding paragraph into “Rationale”?
+
+Rationale
+=========
+
+.. https://mail.python.org/pipermail/python-ideas/2016-August/041727.html
+
+As mentioned, a previous version of PEP 498 allowed escape sequences
+anywhere in f-strings, including as ways to encode the braces delimiting
+the expression portions and in their code.  They would be expanded before
+the code is parsed, which would have had several important ramifications:
+
+#. It would not be clear to human readers which portions are Expressions
+and which are strings.  Great material for an “obfuscated/underhanded
+Python challenge”
+#. Syntax highlighters are good in parsing nested grammar, but not
+in recognizing escape sequences.  ECMAScript 2016 (JavaScript) allows
+escape sequences in its identifiers [#]_ and the author knows of no
+syntax highlighter able to correctly highlight code making use of this.
+
+As a consequence, the expression portions would be harder to recognize
+with and without the aid of syntax highlighting.  With the new grammar,
+it is easy to extend syntax highlighters to correctly parse
+and display f-literals:
+
+.. raw:: html
+
+   <pre><span style=color:#ff5500>f'Magic wand: </span><span style=color:#3daee9>{</span>bag[<span style=color:#bf0303>'wand'</span>]<span style=color:#3daee9>:^10}</span><span style=color:#ff5500>'</span></pre>
+
+.. This is the output of kate-syntax-highlighter when given that code
+   (with some quotes stripped)
+
+Highlighting expression portions with possible escape sequences would
+mean to create a modified copy of all rules of the complete expression
+grammar, accounting for the possibility of escape sequences in key words,
+delimiters, and all other language syntax. One such duplication would
+yield one level of escaping depth and have to be repeated for a deeper
+escaping in a recursive f-literal. This is the case since no highlighting
+engine known to the author supports expanding escape sequences before
+applying rules to a certain context. Nesting contexts however is a
+standard feature of all highlighting engines.
+
+Familiarity also plays a role: Arbitrary nesting of expressions
+without expansion of escape sequences is available in every single
+other language employing a string interpolation method that uses
+expressions instead of just variable names. [#]_
```

Comment:
Do give names to your references, else it's easy to get the sequence wrong if you add a reference later on.